### PR TITLE
Add UUID generation to app details

### PR DESCRIPTION
**Pull request by fredclausen**
This PR adds in the `acars_router_uuid` field to the `app` field and updates the tests to ensure it works.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -282,32 +284,39 @@ impl AppDetails {
...
+    /// let mut vdlm2: AppDetails = AppDetails { name: "dumpvdl2".to_string(), ver: "2.2.0".to_string(), proxied: None, proxied_by: None, acars_router_version: None, acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string())  };
+    /// let manual_acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: Some(true), proxied_by: Some("acars_router".to_string()), acars_router_version: Some("1.0.12".to_string()), acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string()) };
+    /// let mut acars: AppDetails = AppDetails { name: "acarsdec". to_string(), ver: "3.7".to_string(), proxied: None, proxied_by: None, acars_router_version: None, acars_router_uuid: Some("00000000-0000-0000-0000-000000000000".to_string())  };
     /// vdlm2.proxy("acars_router", "1.0.12");
     /// acars.proxy("acars_router", "1.0.12");
     /// assert_eq!(vdlm2, manual_vdlm2);
     /// assert_eq!(acars, manual_acars);
     /// ```
     pub fn proxy(&mut self, proxied_by: &str, acars_router_version: &str) {
+        let acars_router_uuid = match self.acars_router_uuid.is_some() {
```
Comment (jcdeimos) [inline src/lib.rs]: No need to do this, as you can get away with:

```
if self.acars_router_uuid.is_none() {
    self.acars_router_uuid = Some(Uuid::new_v4().to_string());
}
```

Gets rid of the need to do an `.unwrap()` and an extra assignment, and only sets it if it's not set already.

File: src/lib.rs [hunk 2]
```diff
@@ -282,32 +284,39 @@ impl AppDetails {
...
+            true => self.acars_router_uuid.as_ref().unwrap().to_string(),
+            false => Uuid::new_v4().to_string(),
+        };
+
         self.proxied = Some(true);
         self.proxied_by = Some(proxied_by.to_string());
         self.acars_router_version = Some(acars_router_version.to_string());
+        self.acars_router_uuid = Some(acars_router_uuid);
```
Comment (jcdeimos) [inline src/lib.rs]: Can be removed with the change above.

**Review by jcdeimos — CHANGES_REQUESTED** [review]
Can get rid of that `.unwrap()` and assignment and it should be good, provided the tests still pass.

**Comment by fredclausen** [comment]
Should be good to go now, I think.

**Review by jcdeimos — APPROVED** [review]

**Pull request merged**